src/databricks/labs/ucx/progress/workflows.py

```diff
@@ -104,7 +104,21 @@ def crawl_cluster_policies(self, ctx: RuntimeContext) -> None:
     def setup_table_migration(self, ctx: RuntimeContext) -> None:
         """(Optimization) Starts `table_migration` job cluster in parallel to crawling tables."""
 
-    @job_task(depends_on=[crawl_tables, setup_table_migration], job_cluster="table_migration")
+    @job_task(depends_on=[setup_table_migration], job_cluster="table_migration")
+    def verify_prerequisites_table_migration(self, ctx: RuntimeContext) -> None:
+        """Verify the prerequisites for running this job on the table migration cluster are fulfilled:
+        - UC metastore exists
+        - UCX catalog exists.
+        - Assessment workflow ran.
+        """
+        if not (
+            ctx.verify_has_metastore.verify_metastore()
+            and ctx.verify_has_ucx_catalog.verify()
+            and ctx.deployed_workflows.validate_step("assessment")
```

Comment:
This call is interesting, because it will block if the assessment has started but not completed.

As written the behaviour will be:

 - If not yet started, we immediately fail.
 - If pending or running, we will wait up to 20 minutes for the `assessment` run to finish. If it's still not finished, we fail with a timeout. Otherwise…
 - If it finished with success, we're okay.
 - If it didn't finish with success, we're not okay.

This seems fine, but I think the potential for this to block should be noted somewhere.